vignettes/decorate-module-output.Rmd

```diff
@@ -0,0 +1,482 @@
+to these output-modifying objects as decoration objects or decorators.
+
+In below chapter we will present how to create the simplest static decorator with just a server part. Later, we will
+present examples on more advanced usage, where decorators contain UI. You will also learn about a convenience 
+function that makes it easier to write decorators, called [`make_teal_transform_server`](https://insightsengineering.github.io/teal/latest-tagx/reference/make_teal_transform_server.html). The chapter ends with an 
+example module that utilizes decorators and a snippet that uses this module in `teal` application.
+
+### Server
+
+The simplest way to create a decorator is to use `teal_transform_module` with just `server` argument provided (i.e. without UI part).
+This approach adds functionality solely to the server code of the module. 
+In the following example, we assume that the module contains an object (of class `ggplot2`) named `plot`. 
+We modify the title and x-axis label of plot:
+
+```{r}
+library(teal)
+static_decorator <- teal_transform_module(
+  server = function(id, data) {
+    moduleServer(id, function(input, output, session) {
+      reactive({
+        req(data())
+        within(data(), {
+          plot <- plot +
+            ggtitle("This is title") +
+            xlab("x axis")
+        })
+      })
+    })
+  }
+)
+```
+
+To simplify the repetitive elements of writing new decorators 
+(e.g., `function(id, data), moduleServer, reactive, within(data, ...)`), 
+you can use the [`make_teal_transform_server()`](https://insightsengineering.github.io/teal/latest-tag/reference/make_teal_transform_server.html) convenience function, which takes a `language` as input:
+
+```{r}
+static_decorator_lang <- teal_transform_module(
+  server = make_teal_transform_server(
+    expression(
+      plot <- plot +
+        ggtitle("This is title") +
+        xlab("x axis title")
+    )
+  )
+)
+```
+
+### UI
+
+To create a decorator with user interactivity, you can add (optional) UI part and use it in server accordingly (i.e. a typical `shiny` module). 
+In the example below, the x-axis title is set dynamically via a `textInput`, allowing users to specify their preferred label. 
+Note how the input parameters are passed to the [`within`](https://insightsengineering.github.io/teal/latest-tag/reference/teal_data_module.html) function using its `...` argument.
+
+```{r}
+interactive_decorator <- teal_transform_module(
+  ui = function(id) {
+    ns <- NS(id)
+    div(
+      textInput(ns("x_axis_title"), "X axis title", value = "x axis")
+    )
+  },
+  server = function(id, data) {
+    moduleServer(id, function(input, output, session) {
+      reactive({
+        req(data())
+        within(data(),
+          {
+            plot <- plot +
+              ggtitle("This is title") +
+              xlab(x_axis_title)
+          },
+          x_axis_title = input$x_axis_title
```

Comment:
Would it work?

```suggestion
              xlab(my_title)
          },
          my_title = input$x_axis_title
```

I just want to differentiate this and following example. In there - we must use names from `input`.